src/engine/enginebufferscalelinear.cpp

```diff
@@ -249,8 +235,14 @@ CSAMPLE* EngineBufferScaleLinear::do_scale(CSAMPLE* buf,
     ceil_sample[0] = 0;
     ceil_sample[1] = 0;
 
+    int samples_read = 0;
     int i = 0;
-    int screwups = 0;
+    //int screwups_debug = 0;
+
+    double rate_add = fabs(rate_old);
```

Comment:
Yes, it is an issue for your test case, which changes the direction within the loop. 
But we have a special cased this issue above, so that it never happens here. 
And even if we asume we could change the direction, the old impleentation also fails since it iterates only forward.

Thinking about it, It looks like we can drop the spechial case code above and the reverse copyReverse() call if we enable the direction change here. Maybe we can squeze some more CPU sycles. But that is a topic of an other PR.   

I will drop a comment.